### PR TITLE
 Removed chap credentials from the PublishContext

### PR DESCRIPTION
**Pull request by suneeth51**
- Removed chap credentials from the PublishContext
- For CV-CSP, the chap credentials are fetched from the Volume response.
- For Nimble-CSP and 3PAR/Primera-CSP, the chap credentials are retrieved from the `hpeNodeInfos` CRD

### CODE REVIEW

File: pkg/driver/node_server.go [hunk 1]
```diff
@@ -498,21 +534,15 @@ func (driver *Driver) setupDevice(publishContext map[string]string) (*model.Devi
 		DiscoveryIPs:          discoveryIps,
 		ConnectionMode:        defaultConnectionMode,
 		SecondaryArrayDetails: publishContext[secondaryArrayDetailsKey],
-		EncryptionKey: 		   publishContext[hostEncryptionPassphraseKey],
-	}
-	if publishContext[accessProtocolKey] == iscsi {
```
Comment (raunakkumar) [inline pkg/driver/node_server.go:503]: we should still set CHAP only if protocol == iscsi

**Review by raunakkumar — APPROVED** [review]
lgtm

File: pkg/driver/controller_server.go [hunk 2]
```diff
@@ -807,11 +806,6 @@ func (driver *Driver) controllerPublishVolume(
 		return nil, status.Error(codes.NotFound, err.Error())
 	}
 
-	if node.ChapUser != "" && node.ChapPassword != "" {
-		decodedChapPassword, _ := b64.StdEncoding.DecodeString(node.ChapPassword)
```
Comment (raunakkumar) [inline pkg/driver/controller_server.go:811]: Hey @suneeth51 I was just reviewing this with @sneharai4 and we realized that we need to return unencrypted password to CSP.

**Review by raunakkumar — CHANGES_REQUESTED** [review]

File: pkg/driver/controller_server.go (hunk 2 above)
Comment (suneeth51) [reply pkg/driver/controller_server.go:811]: Sure, I will fix that.

**Review by raunakkumar — APPROVED** [review]

**Review by sneharai4 — APPROVED** [review]

**Pull request merged**